lib/paragraph_maker.rb

```diff
@@ -0,0 +1,270 @@
+# frozen_string_literals: true
+
+# Traverse a Nokogiri document tree recursively in order to insert linebreaks.
+module ParagraphMaker
+  extend self
+
+  # Tags whose content we don't touch
+  TAG_NAMES_TO_SKIP = %w[
+    a abbr acronym address audio button dl embed figure h1 h2 h3 h4 h5 h6 hr
+    iframe img input label ol map math noscript object p pre script select
+    source style svg table textarea track video ul
+  ].freeze
+
+  # Tags that need to go inside p tags
+  TAG_NAMES_TO_WRAP = %w[
```

Comment:
> various invalid tags like `style`, `script`, and `noscript` should be skipped and should ignore whitespace nearby, a bunch of other invalid tags like `button`, `input`, `label`, and `textarea` should be wrapped in paragraph tags, etc.

Is this to make sure there's no superfluous paragraphs / breaks around the invalid tags once they're removed by Sanitize?

Reply:
Yeah, that was my attempt to make it so that invalid tags wouldn't produce so many empty paragraph tags. My approach was to skip and ignore whitespace for any tags that would be completely removed, and skip and add paragraphs around most of the inline tags that might leave text behind.

(`input` can't leave text behind, but I put it in the latter category rather than the former because it's more similar to `button` than `script`.)

Reply:
I feel like we shouldn't have a mix of tags allowed by the Sanitize config, and tags that are not. Should we keep them in their own lists and add them into `TAG_NAMES_TO_WRAP` as needed?

In the case of

> any tags that would be completely removed

I think you can use `Sanitize::Config::ARCHIVE[:remove_contents]` (or `Sanitize::Config::DEFAULT[:remove_contents]`).

Reply:
How's this reorganization?

Reply:
Looks great, thanks!